python.d/ntpd.chart.py

```diff
@@ -0,0 +1,383 @@
+        Extracts key=value pairs with float/integer from ntp response packet data.
+        """
+        data = dict()
+        for key, value in self.regex.findall(raw):
+            data[key] = float(value) * PRECISION
+        return data
+
+
+class Peer(Base):
+    def __init__(self, idx, name):
+        self.id = idx
+        self.real_name = name
+        self.name = name.replace('.', '_')
+        self.request = self.get_header(self.id)
+
+    def get_data(self, raw):
+        """
+        Extracts key=value pairs with float/integer from ntp response packet data.
+        """
+        data = dict()
+        for key, value in self.regex.findall(raw):
+            dimension = '_'.join([self.name, key])
+            data[dimension] = float(value) * PRECISION
+        return data
+
+
+class Service(SocketService):
+    def __init__(self, configuration=None, name=None):
+        SocketService.__init__(self, configuration=configuration, name=name)
+        self.order = list(ORDER)
+        self.definitions = dict(CHARTS)
+
+        self.port = 'ntp'
+        self.dgram_socket = True
+        self.system = System()
+        self.peers = dict()
+        self.request = str()
+        self.retries = 0
+        self.show_peers = self.configuration.get('show_peers', False)
+
+    def check(self):
+        """
+        Checks if we can get valid systemvars.
+        If not, returns None to disable module.
+        """
+        self._parse_config()
+
+        if self.show_peers:
+            self.definitions.update(PEER_CHARTS)
+
+            peer_filter = self.configuration.get('peer_filter', r'127\..*')
+            try:
+                self.peer_filter = re.compile(r'^((0\.0\.0\.0)|({0}))$'.format(peer_filter))
+            except re.error as error:
+                self.error('Compile pattern error (peer_filter) : {0}'.format(error))
+                return None
+
+            try:
+                self.peer_rescan = int(self.configuration.get('peer_rescan', peer_rescan))
+                if self.peer_rescan <= 0:
+                    raise ValueError('int > 0 expected: {0}'.format(self.peer_rescan))
+            except ValueError as error:
+                self.error('Value error (peer_rescan) : {0}'.format(error))
+                return None
+
+        self.request = self.system.request
+        raw_systemvars = self._get_raw_data()
+
+        if not self.system.get_data(raw_systemvars):
+            return None
+
+        return True
+
+    def get_data(self):
+        """
+        Gets systemvars data on each update.
+        Gets peervars data for all peers on each update.
+        """
+        data = dict()
+
+        self.request = self.system.request
+        raw = self._get_raw_data()
+        if not raw:
+            return None
+
+        data.update(self.system.get_data(raw))
+
+        if not self.show_peers:
+            return data
+
+        if self.runs_counter == 1 or self.runs_counter % self.peer_rescan or self.retries > 8:
```

Comment:
I suggest to use
```python
if not self.peers or self.runs_counter % self.peer_rescan == 0 or self.retries > 8
```
`not self.peers` - first run
`self.runs_counter % self.peer_rescan == 0` - every self.peer_rescan runs
to run it on every update `self.peer_rescan` should be 1